src/Uno.UI/UI/Xaml/Controls/TreeView/TreeViewItem.cs

```diff
@@ -0,0 +1,776 @@
+			bool handled = false;
+
+			// Inputs for Collapse/Move to parent
+			if ((key == VirtualKey.Left && !flowDirectionReversed) ||
+				(key == VirtualKey.Right && flowDirectionReversed))
+			{
+				if (isExpanded)// Is expanded : need to collapse
+				{
+					targetNode.IsExpanded = false;
+					var targetValue = treeView.ContainerFromNode(targetNode);
+					if (targetValue != null)
+					{
+						var targetItem = (TreeViewItem)targetValue;
+						targetItem.Focus(FocusState.Keyboard);
+					}
+
+					handled = true;
+				}
+				else // Is collapsed: need to select parent
+				{
+					var parentNode = targetNode.Parent;
+					if (parentNode != null)
+					{
+						var parentValue = treeView.ContainerFromNode(parentNode);
+						if (parentValue != null)
+						{
+							var parentItem = (TreeViewItem)parentValue;
+							parentItem.Focus(FocusState.Keyboard);
+							handled = true;
+						}
+					}
+				}
+			}
+			// Inputs for Expand/Move to first child
+			else if ((key == VirtualKey.Right && !flowDirectionReversed) ||
+					 (key == VirtualKey.Left && flowDirectionReversed))
+			{
+				if (!isExpanded && targetNode.HasChildren) // Is collapsed : need to expand
+				{
+					targetNode.IsExpanded = true;
+					handled = true;
+				}
+				else if (targetNode.Children.Count > 0) // Is expanded and has children: need to select first child (if applicable)
+				{
+					var childNode = targetNode.Children[0];
+					var childValue = treeView.ContainerFromNode(childNode);
+					if (childValue != null)
+					{
+						var childItem = (TreeViewItem)childValue;
+						childItem.Focus(FocusState.Keyboard);
+						handled = true;
+					}
+				}
+			}
+
+			return handled;
+		}
+
+		private void OnExpandCollapseChevronPointerPressed(object sender, PointerRoutedEventArgs args)
+		{
+			TreeViewNode targetNode = TreeNode;
+			var isExpanded = !targetNode.IsExpanded;
+			targetNode.IsExpanded = isExpanded;
+			args.Handled = true;
+		}
+
+		private void RecycleEvents(bool useSafeGet = false)
+		{
+			//var chevron = m_expandCollapseChevron;
+			//if (chevron != null)
+			//{
+			//	if (m_expandCollapseChevronPointerPressedToken.value)
+			//	{
+			//		chevron.PointerCanceled(m_expandCollapseChevronPointerPressedToken);
+			//		m_expandCollapseChevronPointerPressedToken.value = 0;
+			//	}
+			//}
+		}
+
+		///* static */
+		private static bool IsDirectionalKey(VirtualKey key)
+		{
+			return
+				key == VirtualKey.Up ||
+				key == VirtualKey.Down ||
+				key == VirtualKey.Left ||
+				key == VirtualKey.Right;
+		}
+
+		private TreeViewNode TreeNode => AncestorTreeView?.NodeFromContainer(this);
+
+		//Setting IsExpanded changes the itemssource collection on the listview, which cannot be done during layout.
+		//We schedule it on the dispatcher so that it runs after layout pass.
+		private void UpdateNodeIsExpandedAsync(TreeViewNode node, bool isExpanded)
+		{
+			var dispatcher = Window.Current.Dispatcher;
+			var ignore = dispatcher.RunAsync(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 'ignore'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)